tests/upgrading/kubernetes_upgrade_test.go

```diff
@@ -0,0 +1,139 @@
+	"github.com/rancher/tfp-automation/framework"
+	cleanup "github.com/rancher/tfp-automation/framework/cleanup"
+	"github.com/rancher/tfp-automation/tests/extensions/provisioning"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type KubernetesUpgradeTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	standardUserClient *rancher.Client
+	session            *session.Session
+	terraformConfig    *config.TerraformConfig
+	clusterConfig      *config.TerratestConfig
+	terraformOptions   *terraform.Options
+}
+
+func (r *KubernetesUpgradeTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	r.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(r.T(), err)
+
+	r.client = client
+
+	enabled := true
+	var testuser = namegen.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(r.T(), err)
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(r.T(), err)
+
+	r.standardUserClient = standardUserClient
+
+	terraformConfig := new(config.TerraformConfig)
+	ranchFrame.LoadConfig("terraform", terraformConfig)
+
+	r.terraformConfig = terraformConfig
+
+	clusterConfig := new(config.TerratestConfig)
+	ranchFrame.LoadConfig("terratest", clusterConfig)
+
+	r.clusterConfig = clusterConfig
+
+	terraformOptions, err := framework.Setup(r.T())
+	require.NoError(r.T(), err)
+
+	r.terraformOptions = terraformOptions
+}
+
+func (r *KubernetesUpgradeTestSuite) TestKubernetesUpgrade() {
+	nodeRolesAll := []config.Nodepool{config.AllRolesNodePool}
+	nodeRolesShared := []config.Nodepool{config.EtcdControlPlaneNodePool, config.WorkerNodePool}
+	nodeRolesDedicated := []config.Nodepool{config.EtcdNodePool, config.ControlPlaneNodePool, config.WorkerNodePool}
+
+	tests := []struct {
+		name      string
+		nodeRoles []config.Nodepool
+		client    *rancher.Client
+	}{
+		{"1 Node all roles " + config.AdminClientName.String(), nodeRolesAll, r.client},
+		{"1 Node all roles " + config.StandardClientName.String(), nodeRolesAll, r.standardUserClient},
+		{"2 nodes - etcd/cp roles per 1 node " + config.AdminClientName.String(), nodeRolesShared, r.client},
+		{"2 nodes - etcd/cp roles per 1 node " + config.StandardClientName.String(), nodeRolesShared, r.standardUserClient},
+		{"3 nodes - 1 role per node " + config.AdminClientName.String(), nodeRolesDedicated, r.client},
+		{"3 nodes - 1 role per node " + config.StandardClientName.String(), nodeRolesDedicated, r.standardUserClient},
+	}
+
+	for _, tt := range tests {
+		clusterConfig := *r.clusterConfig
+		clusterConfig.Nodepools = tt.nodeRoles
+
+		clusterName := namegen.AppendRandomString("tfp")
```

Comment:
nit; this should be in a const, probably in tests/extensions/provisioning since it is referenced by basically every test.

Reply:
I'm not opposed to it, but each package is separate so we would need to do a const block in each package in the test file. Is it that necessary?